src/mbgl/style/expression/format_expression.cpp

```diff
@@ -160,14 +163,25 @@ mbgl::Value FormatExpression::serialize() const {
 EvaluationResult FormatExpression::evaluate(const EvaluationContext& params) const {
     std::vector<FormattedSection> evaluatedSections;
     for (const auto& section : sections) {
-        auto textResult = section.text->evaluate(params);
-        if (!textResult) {
-            return textResult.error();
+        auto contentResult = section.content->evaluate(params);
+        if (!contentResult) {
+            return contentResult.error();
         }
-        
-        optional<std::string> evaluatedText = toString(*textResult);
-        if (!evaluatedText) {
-            return EvaluationError({ "Could not coerce format expression text input to string." });
+
+        optional<std::string> evaluatedText;
+        if (typeOf(*contentResult) == type::Image) {
+            const auto& image = contentResult->get<Image>();
+            // Omit sections with empty image ids.
```

Comment:
why is it not an error?

Reply:
I don't think it is an error, text section could be empty as well as an image similar to `<img src=""></img>`

```json
["format", ["image", ["get", "image_name_from_tile_could_be_empty"]]]
```